pkg/cmd/adm/register_member_test.go

```diff
@@ -589,104 +412,109 @@ func TestRegisterMember(t *testing.T) {
+		tcs := &toolchainv1alpha1.ToolchainClusterList{}
+		require.NoError(t, fakeClient.List(context.TODO(), tcs, runtimeclient.InNamespace("toolchain-host-operator")))
+		assert.Empty(t, tcs.Items)
+		require.NoError(t, fakeClient.List(context.TODO(), tcs, runtimeclient.InNamespace("toolchain-member-operator")))
+		assert.Empty(t, tcs.Items)
+	})
+}
 
-	for _, clusterType := range configuration.ClusterTypes {
-		t.Run("for cluster name: "+clusterType.String(), func(t *testing.T) {
-			t.Run("single toolchain in cluster", func(t *testing.T) {
-				test(t, clusterType, "", false)
-			})
-			t.Run("single toolchain in cluster with letsencrypt", func(t *testing.T) {
-				test(t, clusterType, "", true, "--lets-encrypt")
-			})
-			t.Run("multiple toolchains in cluster", func(t *testing.T) {
-				test(t, clusterType, "asdf", false, "--multi-member", "asdf")
-			})
-			t.Run("multiple toolchains in cluster with letsencrypt", func(t *testing.T) {
-				test(t, clusterType, "42", true, "--multi-member", "42", "--lets-encrypt")
-			})
-		})
+func mockCreateToolchainClusterInNamespaceWithReadyCondition(fakeClient *test.FakeClient, namespace string) {
+	fakeClient.MockCreate = func(ctx context.Context, obj runtimeclient.Object, opts ...runtimeclient.CreateOption) error {
+		if obj, ok := obj.(*toolchainv1alpha1.ToolchainCluster); ok {
+			// force the ready condition on the member toolchaincluster in host cluster only
+			if obj.GetNamespace() == namespace {
+				obj.Status = toolchainv1alpha1.ToolchainClusterStatus{
+					Conditions: []toolchainv1alpha1.Condition{
+						{
+							Type:   toolchainv1alpha1.ConditionReady,
+							Status: corev1.ConditionTrue,
+						},
+					},
+				}
+			}
+		}
+		return fakeClient.Client.Create(ctx, obj, opts...)
 	}
 }
 
-func TestRunAddClusterScriptFailed(t *testing.T) {
-	// given
-	SetFileConfig(t, Host(), Member())
-	hostKubeconfig := PersistKubeConfigFile(t, HostKubeConfig())
-	memberKubeconfig := PersistKubeConfigFile(t, MemberKubeConfig())
-	gock.New(AddClusterScriptDomain).
-		Get(AddClusterScriptPath).
-		Persist().
-		Reply(404)
-	defer gock.OffAll()
-
-	for _, clusterType := range configuration.ClusterTypes {
-		t.Run("for cluster name: "+clusterType.String(), func(t *testing.T) {
-			// given
-			expArgs := []string{"--type", clusterType.String(), "--host-kubeconfig", hostKubeconfig, "--host-ns", "host-ns", "--member-kubeconfig", memberKubeconfig, "--member-ns", "member-ns", "--lets-encrypt"}
-			ocCommandCreator := NewCommandCreator(t, "echo", "bash",
-				AssertFirstArgPrefixRestEqual("(.*)/add-cluster-(.*)", expArgs...))
-			term := NewFakeTerminalWithResponse("Y")
-
-			// when
-			err := runAddClusterScript(term, ocCommandCreator, clusterType, hostKubeconfig, "host-ns", memberKubeconfig, "member-ns", "", true)
-
-			// then
-			require.Error(t, err)
-			assert.NotContains(t, term.Output(), "Command to be called")
-		})
+func mockCreateToolchainClusterWithReadyCondition(fakeClient *test.FakeClient) {
```

Comment:
can you add also a test-case when the ToolchainCluster doesn't become ready? in other words, test-case without this mock?

Reply:
those should be already covered by: https://github.com/kubesaw/ksctl/pull/53/files?diff=unified&w=0#diff-a3205f9a3332e23ec40e20a1c8e099b4ad847b380e2fe3e6829f84344ea0870fR108 - here I'm setting only the host toolchaincluster in member operator to ready, leaving the member toolchaincluster unready

and https://github.com/kubesaw/ksctl/pull/53/files#diff-a3205f9a3332e23ec40e20a1c8e099b4ad847b380e2fe3e6829f84344ea0870fL218 - here I'm setting the member toolchaincluster to ready leaving the hsot toolchaincluster unready

Does this cover the use case you were looking for?

Reply:
yeah, that's it. I missed this one. Thanks